controllers/componentdetectionquery_controller_conditions.go

```diff
@@ -31,59 +31,73 @@ import (
+		return
+	}
+	patch := client.MergeFrom(currentCDQ.DeepCopy())
 
-	patch := client.MergeFrom(componentDetectionQuery.DeepCopy())
-
-	meta.SetStatusCondition(&componentDetectionQuery.Status.Conditions, metav1.Condition{
+	meta.SetStatusCondition(&currentCDQ.Status.Conditions, metav1.Condition{
 		Type:    "Processing",
 		Status:  metav1.ConditionTrue,
 		Reason:  "Success",
 		Message: "ComponentDetectionQuery is processing",
 	})
+	currentCDQ.Spec.GitSource = componentDetectionQuery.Spec.GitSource
+	currentCDQ.Spec.Secret = componentDetectionQuery.Spec.Secret
+	currentCDQ.Status.ComponentDetected = componentDetectionQuery.Status.ComponentDetected
 
-	err := r.Client.Status().Patch(ctx, componentDetectionQuery, patch)
+	err = r.Client.Status().Patch(ctx, &currentCDQ, patch)
 	if err != nil {
 		log.Error(err, "Unable to update ComponentDetectionQuery")
 	}
 }
 
-func (r *ComponentDetectionQueryReconciler) SetCompleteConditionAndUpdateCR(ctx context.Context, req ctrl.Request, componentDetectionQuery *appstudiov1alpha1.ComponentDetectionQuery, originalCDQ *appstudiov1alpha1.ComponentDetectionQuery, completeError error) {
+func (r *ComponentDetectionQueryReconciler) SetCompleteConditionAndUpdateCR(ctx context.Context, req ctrl.Request, componentDetectionQuery *appstudiov1alpha1.ComponentDetectionQuery, completeError error) {
 	log := ctrl.LoggerFrom(ctx)
 
-	patch := client.MergeFrom(originalCDQ.DeepCopy())
-
+	var currentCDQ appstudiov1alpha1.ComponentDetectionQuery
+	err := r.Get(ctx, req.NamespacedName, &currentCDQ)
+	if err != nil {
+		return
+	}
+	copiedCDQ := currentCDQ.DeepCopy()
+	patch := client.MergeFrom(currentCDQ.DeepCopy())
+	condition := metav1.Condition{}
 	if completeError == nil {
-		meta.SetStatusCondition(&componentDetectionQuery.Status.Conditions, metav1.Condition{
+		condition = metav1.Condition{
 			Type:    "Completed",
 			Status:  metav1.ConditionTrue,
 			Reason:  "OK",
 			Message: "ComponentDetectionQuery has successfully finished",
-		})
-		logutil.LogAPIResourceChangeEvent(log, componentDetectionQuery.Name, "ComponentDetectionQuery", logutil.ResourceComplete, nil)
-
+		}
 	} else {
-		meta.SetStatusCondition(&componentDetectionQuery.Status.Conditions, metav1.Condition{
+		condition = metav1.Condition{
 			Type:    "Completed",
 			Status:  metav1.ConditionFalse,
 			Reason:  "Error",
 			Message: fmt.Sprintf("ComponentDetectionQuery failed: %v", completeError),
-		})
-		logutil.LogAPIResourceChangeEvent(log, componentDetectionQuery.Name, "ComponentDetectionQuery", logutil.ResourceComplete, completeError)
+		}
 	}
-	err := r.Client.Status().Patch(ctx, componentDetectionQuery, patch)
+	logutil.LogAPIResourceChangeEvent(log, componentDetectionQuery.Name, "ComponentDetectionQuery", logutil.ResourceComplete, completeError)
+	meta.SetStatusCondition(&currentCDQ.Status.Conditions, condition)
+	currentCDQ.Spec.GitSource = componentDetectionQuery.Spec.GitSource
+	currentCDQ.Spec.Secret = componentDetectionQuery.Spec.Secret
```

Comment:
Why do we want to reassign Spec.GitSource and Spec.Secret, its not like they are changing 🤔

Reply:
emm.  initially was trying to assign back to ensure the infor is not lost. but you are right, spec is not changing & Spec should not be updated.